src/main/java/net/minecraftforge/fmp/client/multipart/MultipartRegistryClient.java

```diff
@@ -0,0 +1,63 @@
+package net.minecraftforge.fmp.client.multipart;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.block.statemap.IStateMapper;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fmp.multipart.IMultipart;
+
+/**
+ * The clientside multipart registry. Allows you to bind a {@link MultipartSpecialRenderer} to a part.
+ *
+ * @see MultipartSpecialRenderer
+ */
+public class MultipartRegistryClient
+{
+    
+    private static Map<ResourceLocation, IStateMapper> specialMappers = new HashMap<ResourceLocation, IStateMapper>();
```

Comment:
NOOOOOOO
Having these maps in the rendering pipeline is a HUGE waste.. there has to be a better way.